src/box/lua/schema.lua

```diff
@@ -744,6 +754,19 @@ local function update_index_parts(format, parts)
                     parts_can_be_simplified = false
                 end
             end
+
+            local first_illegal_index = 3
+            if parts[i].field ~= nil then
+                first_illegal_index = first_illegal_index - 1
+            end
+            if parts[i].type ~= nil then
+                first_illegal_index = first_illegal_index - 1
+            end
+            if parts[i][first_illegal_index] ~= nil then
+                box.error(box.error.ILLEGAL_PARAMS,
+                        "options.parts[" .. i .. "]: unexpected option " .. parts[i][first_illegal_index])
```

Comment:
The only concern I have now:
```
s:create_index('test1', {parts = {{'asd', 2, type='unsigned'}}})
---
- error: 'Illegal parameters, options.parts[1]: unexpected option 2'
...

```
In this input the first option is wrong, but error message says that unexpected option is second. If this is fixable, it would be nice to see proper error message

Reply:
Fixed: now my check happens after field / type validation and the error becomes clearer:
`error: 'Illegal parameters, options.parts[1]: field was not found by name ''asd'''`